### DISCUSSION TITLE
Room Cleaning Mode

### DISCUSSION THREAD

**Discussion by mrrcomp** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#759] [General]
HI
I can only see "Zone" and "pintopoint" using this card? How can I add the "room cleaning option"??

Regards

mrrcomp

**Comment by PiotrMachowski** [comment]
Which integration provided the map to the card?

**Comment by mrrcomp** [comment]
Hi 👋
Thanks for your quick response.  Im using your integration  https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor

Regards

**Reply by PiotrMachowski** [reply]
[Here](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/discussions/317) you have a guide how to set up room cleaning

**Comment by mrrcomp** [comment]
HI
Thanks still having issues


  1.  I added the attribute rooms and I did get the “generate rooms config”
  2.  I removed the “rooms” attribute but now my map is not loading on the app or the card and I don’t see the “show server controls” or the “perform action on” buttons??
My version is 2.2.4


During the process of doing all these configurations all of a sudden the Mi Home app is no longer able to load map? (this is the second time it has happened since I started attempting to work with the map extracton and mapcard??

Would appreciate your insight is it possible that the issue with the robot is as a result of the integrations?

Regards

mrrcomp

**Reply by PiotrMachowski** [reply]
Yes, it is a known problem :(

You can check out this issue: https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/issues/422
